### PR TITLE
chore(stream): retry finding InstanceStreamers while the operation is in progress

### PR DESCRIPTION
**Pull request by efekarakus**
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the Apache 2.0 License.

### CODE REVIEW

File: internal/pkg/aws/cloudformation/stackset/status.go [hunk 1]
```diff
@@ -88,8 +88,3 @@ func (s InstanceStatus) IsFailure() bool {
 func (s InstanceStatus) String() string {
 	return string(s)
 }
-
-// ProgressInstanceStatuses returns a slice of statuses that are InProgress.
-func ProgressInstanceStatuses() []InstanceStatus {
```
Comment (iamhopaul123) [inline internal/pkg/aws/cloudformation/stackset/status.go:93]: :+1:

**Review by iamhopaul123 — APPROVED** [review]

File: internal/pkg/aws/cloudformation/stackset/status.go [hunk 2]
```diff
@@ -66,12 +66,12 @@ func (s InstanceStatus) IsCompleted() bool {
...
 func (s InstanceStatus) InProgress() bool {
-	for _, wanted := range ProgressInstanceStatuses() {
-		if s == wanted {
-			return true
-		}
+	switch s {
+	case instanceStatusPending, instanceStatusRunning:
+		return true
+	default:
+		return false
```
Comment (huanjani) [inline internal/pkg/aws/cloudformation/stackset/status.go:73]: maybe? 
```suggestion
	return instanceStatusPending || instanceStatusRunning:
```

Comment (huanjani) [inline internal/pkg/aws/cloudformation/stackset/status.go]: maybe?
```suggestion
	return s == instanceStatusPending || instanceStatusRunning
```

File: internal/pkg/stream/cfnstackset.go [hunk 3]
```diff
@@ -40,9 +40,11 @@ type StackSetStreamer struct {
 	lastSentOp stackset.Operation
 	curOp      stackset.Operation
 
-	clock   clock
-	rand    func(int) int
-	retries int
+	// Overriden in tests.
```
Comment (huanjani) [inline internal/pkg/stream/cfnstackset.go]: nit:
```suggestion
	// Overridden in tests.
```

**Review by huanjani — APPROVED** [review]
just two nits; otherwise, looks good! 
feel free to remove the DNM label.

**Pull request merged**